rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingPropertyTest.scala

```diff
@@ -1,27 +1,115 @@
 package coop.rchain.rholang.interpreter.accounting
+import java.nio.file.Files
+
+import cats.Id
+import cats.effect.Sync
+import coop.rchain.crypto.hash.Blake2b512Random
+import cats.implicits._
+import coop.rchain.models._
+import coop.rchain.rholang.interpreter.Runtime.{RhoContext, RhoISpace}
+import coop.rchain.rholang.interpreter.errors.OutOfPhlogistonsError
 import coop.rchain.rholang.{PrettyPrinted, ProcGen}
-import coop.rchain.rholang.interpreter.Interpreter
-import coop.rchain.rholang.syntax.rholang_mercury.Absyn.Proc
+import coop.rchain.rholang.interpreter._
+import coop.rchain.rholang.syntax.rholang_mercury.Absyn.{PPar, Proc}
 import coop.rchain.rholang.syntax.rholang_mercury.PrettyPrinter
+import coop.rchain.rspace.history.Branch
+import coop.rchain.rspace.{Context, RSpace}
+import monix.eval.Task
 import org.scalacheck.Arbitrary
 import org.scalacheck.Test.Parameters
 import org.scalatest.{FlatSpec, Matchers}
 import org.scalatest.prop.PropertyChecks
 
+import scala.concurrent.duration._
+import monix.execution.Scheduler.Implicits.global
+
 class CostAccountingPropertyTest extends FlatSpec with PropertyChecks with Matchers {
+  import CostAccountingPropertyTest._
+
   implicit val params: Parameters = Parameters.defaultVerbose.withMinSuccessfulTests(1000)
   implicit val procArbitrary = Arbitrary(
     ProcGen.topLevelGen(5).map(PrettyPrinted[Proc](_, PrettyPrinter.print))
```

Comment:
What happens for bigger levels? Our AST is nested enough that in 5 hops we won't even be able to reach some of the productions...

Reply:
The current implementation has a unlimited branching factor so for bigger heights the chances are the AST grows way too big and sometimes getting an OutOfMemory error.

I'm thinking of adding a "maxBranchFactor" parameter to the generator and that would allow us to better control the shape of the AST.

Reply:
I created this https://rchain.atlassian.net/browse/RHOL-1006